subiquity/server/controllers/ad.py

```diff
@@ -14,14 +14,28 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 import logging
-from typing import List, Optional
+import re
+from typing import List, Optional, Set
 
 from subiquity.common.apidef import API
-from subiquity.common.types import ADConnectionInfo, ADValidationResult
+from subiquity.common.types import (
+    ADConnectionInfo,
+    AdAdminNameValidation,
+    AdDomainNameValidation,
+    AdPasswordValidation
+)
 from subiquity.server.controller import SubiquityController
 
 log = logging.getLogger('subiquity.server.controllers.ad')
 
+DC_RE = r'^[a-zA-Z0-9.-]+$'
+# Validation has changed since Ubiquity. See the following links on why:
+# https://bugs.launchpad.net/ubuntu-mate/+bug/1985971
+# https://github.com/canonical/subiquity/pull/1553#discussion_r1103063195
+# https://learn.microsoft.com/en-us/windows/win32/adschema/a-samaccountname
+AD_ACCOUNT_FORBIDDEN_CHARS = re.escape(r'@"/\[]:;|=,+*?<>')
```

Comment:
I'd prefer if you move the call to `re.escape` where you need it, but it is not a blocker.
```python
AD_ACCOUNT_FORBIDDEN_CHARS = r'@"/\[]:;|=,+*?<>'
[...]
regex = re.compile(f"[{re.escape(AD_ACCOUNT_FORBIDDEN_CHARS)}]")
```